pages/docs/concepts/asyncapi-document/dynamic_channel_names.md

```diff
@@ -0,0 +1,117 @@
+---
+title: Dynamic Channel Names
+weight: 220
+---
+
+In AsyncAPI, dynamic parts of a channel name can be specified using parameters. These parameters are typically defined in the components section of AsyncAPI document. They can be used throughout the document and specifically, they can be referenced in the channels section to denote dynamic parts of channel names.
+
+```mermaid
+graph TD
+  subgraph Components
+    p(Parameters)
+  end
+
+  subgraph Channels
+    c1{Channel 1}
+    c2{Channel 2}
+    c3{Channel 3}
+  end
+
+  p --> c1
+  p --> c2
+  p --> c3
+```
+
+To understand the concept better, let's consider an example where we are modeling a `UserSignedUp` event. This event would involve a channel name like `user/{userId}/signedup`. Here, `{userId}` is a dynamic part, which will vary depending on which user has signed up. You want to define this `{userId}` part as a parameter -
+
+```yml
+components:
+  parameters:
+    userId:
+      description: The ID of the user.
+      schema:
+        type: string
+```
+
+The `userId` parameter is defined in the components section under parameters.
+
+Next, this parameter can be used in the channels section to form the dynamic part of a channel name. It is done using its reference ($ref) -
+
+```yml
+channels:
+  user/{userId}/signedup:
+    parameters:
+      userId:
+        $ref: '#/components/parameters/userId'
+    publish:
+      message:
+        $ref: '#/components/messages/UserSignedUp'
+```
+
+In the above AsyncAPI document, the `user/{userId}/signedup` channel is publishing a `UserSignedUp` message. The `{userId}` in the channel name is replaced dynamically at runtime by the specific ID of the user who signed up.
+
+## Parameter Context
+
+In AsyncAPI, the context of parameters in channel names often refers to where the parameter value comes from. For example, the parameter might be populated from the runtime event payload. This context can be noted in the AsyncAPI document, and it can be used by code generators to make the developer experience easier.
+
+```mermaid
+graph LR
+  subgraph RuntimeEvent
+    e(Event)
+  end
+
+  subgraph Parameters
+    p(Parameter)
+  end
+
+  subgraph Channel
+    c{Channel}
+  end
+
+  e --> p
+  p --> c
+```
+
+Here is an example for parameter context -
+
+```yml
+channels:
+  user/{userId}/signup:
+    parameters:
+      userId:
+        description: "Id of the user."
+        context: "payload"
+        schema:
+          type: "string"
+    subscribe:
```

Comment:
This example is not valid in AsyncAPI v3. Please see https://github.com/asyncapi/website/pull/1868#discussion_r1246407135.

If you need to test your code examples, you can do it manually by following the steps from https://asyncapi.slack.com/archives/C0230UAM6R3/p1689352232934939

Reply:
Thank you for your review @smoya. One thing is for all yml coding example should i add this portion? 

```
asyncapi: 3.0.0
info:
  title: Cool Example
  version: 0.1.0
```

Reply:
This doc should be based on AsyncAPI v3, which is a WIP but the changes on the spec can be read at https://www.asyncapi.com/docs/reference/specification/v3.0.0-next-major-spec.12